fe/fe-core/src/main/java/org/apache/doris/planner/external/iceberg/IcebergScanNode.java

```diff
@@ -334,4 +349,41 @@ public TableIf getTargetTable() {
     public Map<String, String> getLocationProperties() throws UserException {
         return source.getCatalog().getProperties();
     }
+
+    @Override
+    public boolean pushDownAggNoGrouping(FunctionCallExpr aggExpr) {
+        String aggFunctionName = aggExpr.getFnName().getFunction().toUpperCase();
+        return "COUNT".equals(aggFunctionName);
+    }
+
+    @Override
+    public boolean pushDownAggNoGroupingCheckCol(FunctionCallExpr aggExpr, Column col) {
+        return !col.isAllowNull();
+    }
+
+    private long getCountFromSnapshot() {
+        Snapshot snapshot = icebergTable.currentSnapshot();
```

Comment:
Every time to call this `getCountFromSnapshot`, it will return the `current snapshot` of this table. And different call may get different snapshots.
BTW, we support time travel query of iceberg, how to handle it, eg:
`select from iceberg of timestamp xxxx`

Reply:
fixed